src/components/manager/ManagerWorkflowInterfaceContent.js

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useHistory, useLocation } from "react-router-dom";
+import PropTypes from "prop-types";
+
+import { Empty, Form, List, Skeleton, Spin } from "antd";
+const { Item } = Form;
+
+import WorkflowListItem from "./WorkflowListItem";
+
+import { submitIngestionWorkflowRun } from "../../modules/wes/actions";
+
+import DatasetTreeSelect from "./DatasetTreeSelect";
+
+import { withBasePath } from "../../utils/url";
+import StepsTemplate from "./StepsTemplate";
+import WorkflowConfirmationForm, { FIELD_OPTIONS } from "./WorkflowConfirmationForm";
+
+// TODO: when redirected to this page from a Project/Dataset page,  doesn't defaults to selected Project/Dataset,
+// fix: Have the select box input the value and pass that along
+const ManagerWorkflowInterfaceContent = ({ managerType }) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const location = useLocation();
+    console.log("location", location);
+
+    const servicesByID = useSelector((state) => state.services.itemsByID);
+    const datasetsByID = useSelector((state) =>
+        Object.fromEntries(
+            Object.values(state.projects.itemsByID).flatMap((p) => p.datasets.map((d) => [d.identifier, d]))
+        )
+    );
+    const isSubmittingIngestionRun = useSelector((state) => state.runs.isSubmittingIngestionRun);
+
+    const workflows = useSelector((state) =>
+        Object.entries(state.serviceWorkflows.workflowsByServiceID)
+            .filter(([_, s]) => !s.isFetching)
+            .flatMap(([serviceID, s]) =>
+                Object.entries(s.workflows).flatMap(([action, workflowsByAction]) =>
+                    Object.entries(workflowsByAction).map(([id, v]) => ({
+                        ...v,
+                        id, // e.g. phenopacket_json, vcf_gz
+                        serviceID,
+                        action,
+                    }))
+                )
+            )
+    );
```

Comment:
the same ornate code now appears three times in the code base (it appears to begin at workflowsStateToPropsMixin() in proptTypes.js). Do we need it three times?

Reply:
changed it to a utility function